CRM/Streetimport/GP/Handler/TEDIContactRecordHandler.php

```diff
@@ -350,17 +353,51 @@ protected function getDate($record) {
    *
    * @param $contact_id
    * @param $record
+   * @param $parent_activity_id
    *
    * @throws \CiviCRM_API3_Exception
    */
-  public function performContactBaseUpdates($contact_id, $record) {
+  public function performContactBaseUpdates($contact_id, $record, $parent_activity_id) {
     //Contact Entity
     $contact_params = $this->getPreparedContactParams($record);
     $this->updateContact($contact_params, $contact_id, $record);
 
+    // Selection date
```

Comment:
I'd like to move this whole check into `createOrUpdateAddress()` instead, for a few reasons:
- `createOrUpdateAddress()` currently already deals with a few scenarios where an address is either updated/created, or a MUR activity is created instead. This would be another scenario
- `createOrUpdateAddress()` also performs other checks on addresses (e.g. validity according to `de.systopia.postcodeat` data). This is displayed as part of the MUR activity, but that won't be the case here if we have conflicting changes that trigger the MUR activity in `performContactBaseUpdates()`
- There are a few scenarios in `createOrUpdateAddress()` where we discard updates; this will no longer work if the address change check is triggered first

This might be a good time to also refactor the code in `createOrUpdateAddress` a bit. Currently, it deals with three scenarios, and for some scenarios there's a lot of repeating code with slight variations:
1. Address is created or updated immediately
2. Address change triggers a MUR (scenarios: invalid address, incomplete address, address changed since (new))
3. Address change is discarded

The code especially especially for 2. can be simplified quite a bit by separating the checks from the MUR creation, potentially passing multiple failed checks to the template (i.e. you could have a scenario where the new address is invalid, and at the same time there's been an address change since the selection => both facts should be included in the MUR activity).

(We'd eventually like to store these MURs in a more structured way - so that the user can just select "Apply/Discard" to update the address rather than having to copy/paste stuff, so refactoring this code will make that easier down the line.)